src/mlang/backend_compilers/dgfip_gen_files.ml

```diff
@@ -130,27 +130,37 @@ let is_input st = match st with Base | Computed -> false | _ -> true
 let is_computed st = match st with Base | Computed -> true | _ -> false
 
 let input_var_subtype iv : var_subtype =
-  match Pos.unmark iv.Mast.input_subtyp with
-  | Mast.Context -> Context
-  | Family -> Family
-  | Penality -> Penality
-  | Income -> Income
+  List.find_map
+    (fun t ->
+      match Pos.unmark t with
+      | "contexte" -> Some Context
+      | "famille" -> Some Family
+      | "penalite" -> Some Penality
+      | "revenu" -> Some Income
+      | _ -> None)
+    iv.Mast.input_category
+  |> function
+  | Some s -> s
+  | None -> assert false
```

Comment:
So the issue here would be that none of the expected categories appears, not that there is an unexpected one. I definitely would add a clean error message here, but this file in its current state ought to fall into oblivion as it's mostly adhoc, hardcoded, necessary mess that works only because tailored for the current M codebase.